### PR TITLE
Feature: get_program_price

### PR DESCRIPTION
**Pull request by 1yam**
Problem:
In the Pay-As-You-Go (PAYG) flow, we need to retrieve the price of a program.

Solution:
Implement the get_program_price function to query the `/api/v0/price/` endpoint and fetch the price of the program.

### CODE REVIEW

File: src/aleph/sdk/client/http.py [hunk 1]
```diff
@@ -392,3 +397,17 @@ async def watch_messages(
                         yield parse_message(data)
                 elif msg.type == aiohttp.WSMsgType.ERROR:
                     break
+
+    async def get_program_price(self, item_hash: str) -> PriceResponse:
```
Comment (hoh) [inline src/aleph/sdk/client/http.py:401]: What guarantees do we have that the `item_hash` corresponds to a program ?

File: src/aleph/sdk/client/http.py [hunk 2]
```diff
@@ -392,3 +397,17 @@ async def watch_messages(
...
+        async with self.http_session.get(f"/api/v0/price/{item_hash}") as resp:
+            try:
+                resp.raise_for_status()
+                response_json = await resp.json()
+                return PriceResponse(
+                    required_tokens=response_json["required_tokens"],
+                    payment_type=response_json["payment_type"],
+                )
+            except aiohttp.ClientResponseError as e:
+                if e.status == 400:
```
Comment (hoh) [inline src/aleph/sdk/client/http.py:411]: 400 == Bad Request, should this not be 404 ?

File: tests/unit/test_price.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+from aleph.sdk.query.responses import PriceResponse
+
+
+def make_mock_price_client(item_hash: str):
+    mock_client = AsyncMock(spec=AlephHttpClient)
+    if item_hash == "valid_item_hash":
+        mock_client.get_program_price.return_value = PriceResponse(
+            required_tokens=3.0555555555555556e-06, payment_type="superfluid"
+        )
+    elif item_hash == "invalid_item_hash":
```
Comment (hoh) [inline tests/unit/test_price.py:16]: Why not two different mocks ?

File: tests/unit/test_price.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        mock_client.get_program_price.side_effect = InvalidHashError("Invalid hash")
+    else:
+        raise NotImplementedError(f"Mock not implemented for item_hash: {item_hash}")
+    return mock_client
+
+
+@pytest.mark.asyncio
+async def test_get_program_price_valid():
```
Comment (hoh) [inline tests/unit/test_price.py:24]: What is this test testing ? Can you add a docstring ? Would it not be better to mock the response of the HTTP API ?

File: src/aleph/sdk/client/http.py (hunk 2 above)
Comment (1yam) [reply src/aleph/sdk/client/http.py:411]: I made it based on what api is returning me, and it give me a 400.
```
aiohttp.client_exceptions.ClientResponseError: 400, message='Bad Request', url=URL('https://api2.aleph.im/api/v0/price/b')
```

File: src/aleph/sdk/client/http.py (hunk 1 above)
Comment (1yam) [reply src/aleph/sdk/client/http.py:401]: We don't really have one, but in all case it's gonna check server side,
So is it making sense to check if it an executable message one time  on client side and one time on server side ?

**Review by olethanh — APPROVED** [review]

**Pull request merged**